packages/orbit-components/src/Dialog/index.tsx

```diff
@@ -150,51 +52,86 @@ const Dialog = ({
     }
   };
 
-  React.useEffect(() => {
-    const handleKeyDown = ev => {
-      if (ev.keyCode === KEY_CODE_MAP.ESC && onClose) {
-        onClose();
-      }
-    };
-    window.addEventListener("keydown", handleKeyDown);
-    return () => window.removeEventListener("keydown", handleKeyDown);
-  }, [onClose]);
+  const handleKeyDown = (ev: React.KeyboardEvent<HTMLDivElement>) => {
+    if (ev.keyCode === KEY_CODE_MAP.ESC && onClose) {
+      onClose();
+    }
+  };
 
   const dialogID = useRandomId();
+  const descriptionId = useRandomId();
 
   const dialog = (
-    <StyledDialog
+    <div
       ref={wrapperRef}
       data-test={dataTest}
+      onKeyDown={handleKeyDown}
       id={id}
       onClick={handleClose}
       tabIndex={0}
-      role="dialog"
-      aria-modal="true"
-      aria-labelledby={dialogID}
+      role="button"
```

Comment:
I'm sorry but I couldn't find a reference for that, can you please point me to that? 🙏  It is not clear to me why the whole dialog wrapper should have `role="button"`

I am in favor of fixing accessibility issues we might have had in the past, but this one is not clear to me

Reply:
> It is not clear to me why the whole dialog wrapper should have role="button"

Because the whole wrapper is clickable (it's just overlay) and it has to have appropriate role and onClick and onKeyDown, it's ok to have it as button. If the main wrapper will have role dialog (it's non interactive role) it will have a warning from [eslint a11y rule
](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/3d1d26d89d492947cbf69f439deec9e7cfaf9867/docs/rules/no-noninteractive-element-interactions.md) 

For dialog itself is important to have:
- container that serves as dialog main wrapper should have role="dialog"
- some element inside should be focused after it's opened (autofocus)
- aria-label and aria-describedby 

Now it's missing the autofocus on opened, I think we can autofocus the primary action, wdyt? 

In general, I think if the keyboard navigation is ok and screen reader reads correctly everything, then everything should be 👌🏻

Reply:
Yep, I found it [here](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/) and it says that the main content of dialog should have role dialog. In the previous implementation it was throwing a warning, as Dialog can't be interactive element (having onClick and etc), but you also have a fair point, so I slightly adjusted that to have a common wrapper with only role and aria attributes.

Reply:
The another thing I am thinking about is that probably we should autofocus the primary action, when it's opening.

Reply:
My concern is about the `role="button"` on the whole Dialog, which I think is not correct. It is saying the whole dialog + overlay is clickable, which is not true. It also applies (by preflight rules) a cursor pointer to the whole dialog and overlay, which are also undersired